x-pack/plugins/ml/server/lib/alerts/jobs_health_service.ts

```diff
@@ -0,0 +1,173 @@
+import { MlClient } from '../ml_client';
+import {
+  AnomalyDetectionJobsHealthRuleParams,
+  JobSelection,
+} from '../../routes/schemas/alerting_schema';
+import { datafeedsProvider, DatafeedsService } from '../../models/job_service/datafeeds';
+import { ALL_JOBS_SELECTION } from '../../../common/constants/alerts';
+import { MlJobsResponse } from '../../../common/types/job_service';
+import { DatafeedStats } from '../../../common/types/anomaly_detection_jobs';
+import { GetGuards } from '../../shared_services/shared_services';
+import { AnomalyDetectionJobsHealthAlertContext } from './register_jobs_monitoring_rule_type';
+import { getResultJobsHealthRuleConfig } from '../../../common/util/alerts';
+
+interface TestResult {
+  name: string;
+  context: AnomalyDetectionJobsHealthAlertContext;
+}
+
+type TestsResults = TestResult[];
+
+type NotStartedDatafeedResponse = Array<DatafeedStats & { job_id: string }>;
+
+export function jobsHealthServiceProvider(
+  mlClient: MlClient,
+  datafeedsService: DatafeedsService,
+  logger: Logger
+) {
+  /**
+   * Extract result list of job ids based on included and excluded selection of jobs and groups.
+   * @param includeJobs
+   * @param excludeJobs
+   */
+  const getResultJobIds = async (includeJobs: JobSelection, excludeJobs?: JobSelection | null) => {
+    const jobAndGroupIds = [...(includeJobs.jobIds ?? []), ...(includeJobs.groupIds ?? [])];
+
+    const includeAllJobs = jobAndGroupIds.some((id) => id === ALL_JOBS_SELECTION);
+
+    // Extract jobs from group ids and make sure provided jobs assigned to a current space
+    const jobsResponse = (
+      await mlClient.getJobs<MlJobsResponse>({
+        ...(includeAllJobs ? {} : { job_id: jobAndGroupIds.join(',') }),
+      })
+    ).body.jobs;
+
+    let resultJobIds = jobsResponse.map((v) => v.job_id);
+
+    if (excludeJobs && (!!excludeJobs.jobIds.length || !!excludeJobs?.groupIds.length)) {
+      const excludedJobAndGroupIds = [
+        ...(excludeJobs?.jobIds ?? []),
+        ...(excludeJobs?.groupIds ?? []),
+      ];
+      const excludedJobsResponse = (
+        await mlClient.getJobs<MlJobsResponse>({
+          job_id: excludedJobAndGroupIds.join(','),
+        })
+      ).body.jobs;
+
+      const excludedJobsIds: Set<string> = new Set(excludedJobsResponse.map((v) => v.job_id));
+
+      resultJobIds = resultJobIds.filter((v) => !excludedJobsIds.has(v));
+    }
+
+    return resultJobIds;
+  };
+
+  return {
+    /**
+     * Gets not started datafeeds for provided jobs selection.
+     * @param includeJobs
+     * @param excludeJobs
+     */
+    async getNotStartedDatafeed(
+      includeJobs: JobSelection,
+      excludeJobs?: JobSelection | null
+    ): Promise<NotStartedDatafeedResponse | void> {
+      const jobIds = await getResultJobIds(includeJobs, excludeJobs);
+      logger.debug(`Performing health checks for job ids: ${jobIds.join(', ')}`);
+      const dataFeeds = await datafeedsService.getDatafeedByJobId(jobIds);
```

Comment:
should be `datafeeds`

Reply:
Renamed in https://github.com/elastic/kibana/pull/106084/commits/b08031efbc265847be7d3617ca90253110b74fc9